R/vector.R

```diff
@@ -133,11 +134,59 @@ modify <- function(.x, ..., .elts = list()) {
 #' # that's not the case for rlang's aliases:
 #' na_dbl <- NA
 #' typeof(na_dbl)
+#'
+#' # are_na() is vectorised and works regardless of the type
+#' are_na(c(1, 2, NA))
+#' are_na(c(1L, NA, 3L))
+#'
+#' # is_na() checks for scalar input and works for all types
+#' is_na(NA)
+#' is_na(na_dbl)
+#' is_na(character(0))
+#'
+#' # There are typed versions as well:
+#' is_lgl_na(NA)
+#' is_lgl_na(na_dbl)
 #' @name missing
 NULL
 
 #' @rdname missing
 #' @export
+are_na <- is.na
```

Comment:
I remember that one of the challenges of learning R was to figure out what's vectorised and what's not. I think `are_na()` would be beneficial to newcomers.

`is_logical()` and `is_scalar_logical()` are predicates about vectors that can themselves be scalar or vectors. But they both are scalar predicates, unlike `are_na()`. Then, `is_na()` is akin to `is_null()`. And it provides additional intent and type check when you're expecting `NA` as an input.